lib/sqlalchemy/connectors/pyodbc.py

```diff
@@ -77,6 +77,9 @@ def create_connect_args(self, url: URL) -> ConnectArgsType:
         else:
 
             def check_quote(token: str) -> str:
+                # token is already quoted, so return it as-is
+                if token.startswith("{") and token.endswith("}"):
```

Comment:
Note: this line does _not_ perform a type-cast (`str(token)`) unlike the existing conditional.

Reply:
(now I understand that it probably should be, though, since a port number, for example, could be passed as a non-string value)